### ISSUE TITLE
[Talent] Hack and Slash - Damage coming from Offhand aswell.

### ISSUE THREAD

**Issue by crulinjr** [WoWManiaUK/Redemption#3274]
How it is working: Hack and Slash (http://www.wow-mania.com/armory/?spell=13964) is dealing damage with the Offhand. 

How it should work: Hack and Slash damage should ONLY come from the Main Hand weapon. 
https://wowwiki.fandom.com/wiki/Hack_and_Slash.

Obs: As you can see on the image the Min hit is 449 when my Main Hand weapon range is between 1315 - 1690. I have 53.66% ArP. The Max hit is 1327. So it is indeed dealing damage from the mh AND the oh weapon.
![WoWScrnShot_051419_181011](https://user-images.githubusercontent.com/40045543/57751744-512f0800-76ac-11e9-87ed-4311b676e3b6.jpg)

**Comment by volkangul** [comment]
Agree, Extra attacks only deals damage from Main Hand Weapon.

**Comment by Keader** [comment]
The method that handle extra attack is:

![image](https://user-images.githubusercontent.com/7120913/57929862-cb1ce800-788a-11e9-8a11-e3437f2af299.png)

The 2th param of ```AttackerStateUpdate``` is attack type.
To Hack and Slash is always BASE_ATTACK, that means: "Basic Main Hand Weapon".
To be Off Hand, it need be ```OFF_ATTACK``` 
So... nothing wrong here, it always do damage based in main hand.

**Comment by crulinjr** [comment]
I have shown you the proves that this is not true. There is no way Hack and
Slash can do 449 HIT dmg  with my Main Hand having a greater damage range
which is shown in the screenshot. However, if those are in fact MH hits
then something else is lowering the dmg by a lot. Remember i have over 50%
ArP so its not armor.

**Comment by crulinjr** [comment]
Im sorry you got mad at dranah for correcting you publicly but it has
nothing to do that we are working to get the server bug free and working as
blizzlike as possible. Here there is no way you can argue that 449 hit from
Hack and Slash comes from the Main Hand.

**Comment by crulinjr** [comment]
I was reading about Hack and Slash and found this threat from another
server.
https://github.com/TrinityCore/TrinityCore/issues/6892  They had the same
issue.

"Then are you sure http://wotlk.openwow.com/?spell=66923 is actually
modified by m_extraAttacks? Or perhaps the problem lies in BASE_ATTACK
itself, not its relation to m_extraAttacks? Something is messed up, cant
seem to figure out what..."

"Okay after more testing, ive found out that the problem idd lies in
BASE_ATTACK. Currently, BASE_ATTACK DOES NOT act as mainhand weapon attack,
its just an attack with ANY of your weapons. You can see the proof here -
http://oi50.tinypic.com/2mza1d3.jpg . Ive had both titan's grip for dual
wield and warrior's sword spec from arms tree. Sword spec procs could be
offhand attacks. So in the end, BASE_ATTACK needs to be fixed"

"Still bugged. Could the problem be somewhere within this part of the code?
{
case BASE_ATTACK:
damageInfo->procAttacker = PROC_FLAG_DONE_MELEE_AUTO_ATTACK |
PROC_FLAG_DONE_MAINHAND_ATTACK;
damageInfo->procVictim = PROC_FLAG_TAKEN_MELEE_AUTO_ATTACK;
break;
case OFF_ATTACK:
damageInfo->procAttacker = PROC_FLAG_DONE_MELEE_AUTO_ATTACK |
PROC_FLAG_DONE_OFFHAND_ATTACK;
damageInfo->procVictim = PROC_FLAG_TAKEN_MELEE_AUTO_ATTACK;
damageInfo->HitInfo = HITINFO_OFFHAND;
break;
default:
return;
}"

**Comment by Dranah** [comment]
I did not correct him, I asked to not use dev hax in testing. Is that the end of the world?

PS: I made a test that proves that the damage scales with offhand atm. I understand the code is right, but the code just doesn't do what it should . Lookie:

1) Test with offhand weapon unequipped (theres no offhand weapon = no HNS procs from offhand):
HNS damage matches the damage (min,max,avg) of main hand melee attacks.
Melee (main-hand only): 164-274, avg:222
HnS (main hand only): 176-268, avg: 220

2) Test with equipping a lvl 10 white offhand axe (no stats) just to add the offhand procs to the picture:
The main hand damage does not change: 164-274, but the melee attacks now have an offhand component hitting for 40-80 damage and these OH attacks are capable of proccing HnS too.
The HnS procs damage however, assuming they only scale with main hand, should be unchanged.
But the test shows else:
Melee (MH and OH mixed): 54-253, avg:162
HnS damage (from both MH and OH procs): 54-236, avg:122

The test clearly shows that having the spell capable to proc off an offhand attack results significantly lower damage (with unchanged main hand damage), which is only possible if it calculates with offhand damage too.

**Comment by volkangul** [comment]
A note, HnS damage " SHOULD NOT " be glancing damage since it must come from Main Hand.  " Glancing " damage only comes from "Off-Hand" attacks which proofs HnS damage comes where it proccs from.
We are reporting our findings, we are not expecting you be the same as Retail. But if you are unable to solve an issue just let us know may be we can change our Weapon Spec.

**Comment by Keader** [comment]
Easy test.
Go out of melee range.
Make a macro using: .cast 66923
It is the spell casted by the talent when it procs.

As you can see, damage is always using main weapon damage.

**Comment by crulinjr** [comment]
Im getting a msg saying "there is no such subcommand"

**Comment by crulinjr** [comment]
Im uploading another screenshot where you can see i get a Hack and Slash
HIT of 440.No way my MAIN HAND can do that low damage when the MAX HIT DMG
is 1377. Hack and Slash procs that comes from Melee attacks are also doing
damage with the OFFHAND.

**Comment by Dranah** [comment]
After further intense testing, I have found the root of the issue. Once more, lying in visual display of damage meters. Sigh.
Here's the combat log of a random proc:
![WoWScrnShot_052019_215158](https://user-images.githubusercontent.com/31132743/58047994-db161f80-7b49-11e9-8424-14b984d136fe.jpg)
Clearly the proc has been done by the main hand attack for 168 damage.
What the recount displayed was however a "Melee ( Hack and Slash) : 56 damage"
It has jumped over that line in the log and displayed the next.

I have watched another proc
![WoWScrnShot_052019_215229](https://user-images.githubusercontent.com/31132743/58048057-0ef14500-7b4a-11e9-855e-1025d1ca92c4.jpg)
Same happened, proc for 260ish damage, the next line offhand hit for 55 was copied into recount as "Melee(Hack and Slash)"

Your hack and slash is doing proper damage, always with MH. But recount is copying wrong line into its display so it "looks like" you are getting offhand damage procs.
